File: frontend/src/app/models/accountRequest.ts
export class AccountRequestDTO {
  private name: string;
  private email: string;
  private password: string;
  private confirmPassword: string;

  constructor(name: string,
              email: string,
              password: string,
              confirmPassword: string)
  {
    this.name = name;
    this.email = email;
    this.password = password;
    this.confirmPassword = confirmPassword;
  }
}

File: backend/Localize.Domain/DTOs/Request/AccountRequest.cs
using System.ComponentModel.DataAnnotations;

namespace Localize.Domain.DTOs.Request
{
    public class AccountRequest
    {
        [Required(ErrorMessage = "Nome deve ser preenchido")]
        public string Name { get; set; }
        [Required(ErrorMessage = "Email deve ser preenchido")]
        public string Email { get; set; }
        [Required(ErrorMessage = "Senha deve ser preenchida")]
        public string Password { get; set; }
        [Required(ErrorMessage = "Confirmação de senha deve ser preenchida")]
        public string ConfirmPassword { get; set; }

        public void HashPassword(string password)
        {
            Password = password;
        }
    }
}
